cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +347,131 @@ std::vector<unsigned char> save_model(ModelHandle model) {
+           tree_from_individual_forest.num_nodes,
+         "Error! Mismatch the number of nodes present in a tree in the "
+         "concatenated forest and"
+         " the tree present in the individual forests");
+
+  for (int each_node = 0; each_node < tree_from_concatenated_forest.num_nodes;
+       each_node++) {
+    tl::Tree::Node& node_from_concat = tree_from_concatenated_forest[each_node];
+    tl::Tree::Node& node_from_indiv = tree_from_individual_forest[each_node];
+    ASSERT(node_from_concat.is_root() == node_from_indiv.is_root(),
+           "Error! root position mismatch between concatenated forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.parent() == node_from_indiv.parent(),
+           "Error! node parent mismatch between concatenated forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.is_leaf() == node_from_indiv.is_leaf(),
+           "Error! mismatch in the position of a leaf between concatenated "
+           "forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.leaf_value() == node_from_indiv.leaf_value(),
+           "Error! leaf value mismatch between concatenated forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.cright() == node_from_indiv.cright(),
+           "Error! mismatch in the position of the node between concatenated "
+           "forest and the"
+           " individual forests ");
+    ASSERT(node_from_concat.cleft() == node_from_indiv.cleft(),
+           "Error! mismatch in the position of the node between concatenated "
+           "forest and the"
+           " individual forests ");
+    ASSERT(
+      node_from_concat.split_index() == node_from_indiv.split_index(),
+      "Error! split index value mismatch between concatenated forest and the"
+      " individual forests ");
+  }
+}
+
+/**
+ * @brief Compares the concatenated treelite model with the information of the forest
+ *   present in the different workers
+ * @param[in] concat_tree_handle: ModelHandle for the concatenated forest.
+ * @param[in] treelite_handles: List containing ModelHandles for the forest present in
+ *   each worker.
+ */
+void compare_concat_forest_to_subforests(
+  ModelHandle concat_tree_handle, std::vector<ModelHandle> treelite_handles) {
+  size_t concat_forest;
+  size_t total_num_trees = 0;
+  std::cout << "concat_tree_handle " << concat_tree_handle << std::flush
+            << std::endl;
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    size_t num_trees_each_forest;
+    TREELITE_CHECK(TreeliteQueryNumTree(treelite_handles[forest_idx],
+                                        &num_trees_each_forest));
+    total_num_trees = total_num_trees + num_trees_each_forest;
+  }
+
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_tree_handle, &concat_forest));
+  std::cout << "concat_mod_ num trees " << concat_forest << std::flush
```

Comment:
Seems like this was a debug print? I would leave it out since the rest of the function works via ASSERTS.